src/services/exchange-consumer.ts

```diff
@@ -0,0 +1,91 @@
+import { AmqpConnectionManager, ChannelWrapper } from 'amqp-connection-manager'
```

Comment:
there is no need to design `ExchangeConsumer` and split `Consumer` to two.


you can refer to `spring-amqp` projects `@RabbitListener` usage
[https://www.baeldung.com/rabbitmq-spring-amqp](https://www.baeldung.com/rabbitmq-spring-amqp).

listener method alway consumes from a queue.